mailbox/api/src/main/java/org/apache/james/mailbox/model/Attachment.java

```diff
@@ -73,19 +78,57 @@ private AttachmentId attachmentId() {
             }
             return AttachmentId.random();
         }
+    }
+
+    public static class WithBytes {
+        private final byte[] bytes;
+        private final Attachment metadata;
+
+        public WithBytes(byte[] bytes, Attachment metadata) {
+            Preconditions.checkArgument(bytes != null);
+            this.bytes = bytes;
+            this.metadata = metadata;
+        }
+
+        public Attachment getMetadata() {
+            return metadata;
+        }
+
+        public InputStream getStream() {
```

Comment:
I not a lot of people will agree with me, but i would prefer returning `ByteArrayInputStream` this way we know we don't have to worry about closing it.

Reply:
No I do agree with you